### PR TITLE
CIV-0000 DTSCCI-208: Multiple Incidents: Defendant Can't link claim

### PR DESCRIPTION
**Pull request by srochani**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/app/auth/launchdarkly/launchDarklyClient.ts [hunk 1]
```diff
@@ -38,7 +38,8 @@ export async function isPcqShutterOn(): Promise<boolean> {
 }
 
 export async function isCUIReleaseTwoEnabled(): Promise<boolean> {
-  return await getFlagValue('cuiReleaseTwoEnabled') as boolean;
+  //return await getFlagValue('cuiReleaseTwoEnabled') as boolean;
```
Comment (linusnorton) [inline src/main/app/auth/launchdarkly/launchDarklyClient.ts:41]: remove this if it's not needed

File: src/main/app/client/civilServiceClient.ts [hunk 2]
```diff
@@ -221,6 +222,20 @@ export class CivilServiceClient {
     }
   }
 
+  async isOcmcDefendantLinked(caseReference: string): Promise<boolean> {
+    try {
+      const response = await this.client.get(CIVIL_SERVICE_CHECK_OCMC_DEFENDENT_LINKED_URL //nosonar
+        .replace(':caseReference', caseReference), {headers: {'Content-Type': 'application/json'}});// no-sonar
+      if (!response.data) {
+        return null;
```
Comment (linusnorton) [inline src/main/app/client/civilServiceClient.ts:230]: this shouldn't even compile - needs to be true or false

Comment (srochani) [reply src/main/app/client/civilServiceClient.ts:230]: Its compile, what is the issue here?

File: src/main/app/auth/launchdarkly/launchDarklyClient.ts (hunk 1 above)
Comment (srochani) [reply src/main/app/auth/launchdarkly/launchDarklyClient.ts:41]: Its local change, I will revert it back

File: src/main/app/client/civilServiceClient.ts [hunk 3]
```diff
@@ -221,6 +222,20 @@ export class CivilServiceClient {
...
+  async isOcmcDefendantLinked(caseReference: string): Promise<boolean> {
+    try {
+      const response = await this.client.get(CIVIL_SERVICE_CHECK_OCMC_DEFENDENT_LINKED_URL //nosonar
+        .replace(':caseReference', caseReference), {headers: {'Content-Type': 'application/json'}});// no-sonar
+      if (!response.data) {
+        return false;
+      }
+      return response.data as boolean;
+    } catch (err: unknown) {
+      logger.error('Error when checking a claim is linked to a defendant');
```
Comment (linusnorton) [inline src/main/app/client/civilServiceClient.ts:234]: add case reference

**Review by linusnorton — APPROVED** [review]

File: src/main/app/client/civilServiceUrls.ts [hunk 4]
```diff
@@ -19,6 +19,7 @@ export const CIVIL_SERVICE_AGREED_RESPONSE_DEADLINE_DATE =`${CIVIL_SERVICE_CASES
 export const CIVIL_SERVICE_USER_CASE_ROLE = `${CIVIL_SERVICE_CASES_URL}:claimId/userCaseRoles`;
 export const CIVIL_SERVICE_COURT_DECISION =  `${CIVIL_SERVICE_CASES_URL}:claimId/courtDecision`;
 export const CIVIL_SERVICE_VALIDATE_OCMC_PIN_URL = `${CIVIL_SERVICE_ASSIGNMENT_URL}/reference/:caseReference/ocmc`;
+export const CIVIL_SERVICE_CHECK_OCMC_DEFENDENT_LINKED_URL = `${CIVIL_SERVICE_ASSIGNMENT_URL}/reference/:caseReference/ocmc`;
```
Comment (neeta-hmcts) [inline src/main/app/client/civilServiceUrls.ts:22]: line 21 and line 22 URLs are the same. Any specific reason you created a new URL const?

**Review by neeta-hmcts — APPROVED** [review]

**Review by sankhajuria — DISMISSED** [review]

**Comment by srochani** [comment]
reopen

**Review by nigeldunne — APPROVED** [review]

File: charts/civil-citizen-ui/values.preview.template.yaml [hunk 5]
```diff
@@ -29,7 +29,7 @@ civil-service:
   java:
     applicationPort: 4000
     releaseNameOverride: ${SERVICE_NAME}-civil-service
-    image: 'hmctspublic.azurecr.io/civil/service:latest'
+    image: 'hmctspublic.azurecr.io/civil/service:pr-4608'
```
Comment (velimd) [inline charts/civil-citizen-ui/values.preview.template.yaml:32]: This needs to be reverted before merging no?

File: src/main/app/client/civilServiceClient.ts [hunk 6]
```diff
@@ -248,6 +249,20 @@ export class CivilServiceClient {
...
   }
 
+  async isOcmcDefendantLinked(caseReference: string): Promise<boolean> {
+    try {
+      const response = await this.client.get(CIVIL_SERVICE_CHECK_OCMC_DEFENDENT_LINKED_URL //nosonar
+        .replace(':caseReference', caseReference), {headers: {'Content-Type': 'application/json'}});// no-sonar
+      if (!response.data) {
+        return false;
+      }
+      return response.data as boolean;
```
Comment (velimd) [inline src/main/app/client/civilServiceClient.ts:259]: I believe you can do
`return !!response.data` insteaad of returning as boolean

Comment (velimd) [reply src/main/app/client/civilServiceClient.ts:259]: ignore, saw the backend and the response being returned as a boolean.

**Review by nigeldunne — APPROVED** [review]

**Review by velimd — APPROVED** [review]

**Comment by srochani** [comment]
closing and reopening the PR

**Pull request merged**